src/components/CountDown/CountDown.tsx

```diff
@@ -0,0 +1,194 @@
+      setAppStateVisible(appState.current)
+    })
+    return () => {
+      subscription.remove()
+    }
+  }, [])
+
+  useEffect(() => {
+    const timeLoop = intervalTimeBySecond - new Date().getMilliseconds()
+    const timeout = setTimeout(() => {
+      if (seconds >= 0) {
+        const uff = timeEnd.current - new Date().getTime()
+        setSeconds(uff / milisecond)
+      }
+      if (seconds < 1) {
+        if (loop) {
+          timeEnd.current = timeEnd.current - new Date().getTime()
+          setSeconds(initialSeconds)
+        } else {
+          clearTimeout(timeout)
+          if (onFinish) {
+            onFinish()
+          }
+        }
+      }
+    }, timeLoop)
+    return () => {
+      clearTimeout(timeout)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [seconds, appStateVisible])
+
+  const renderColons = () => (
+    <Container>
+      <Label style={[CountDownTheme.textStyle, textStyle, colonsStyle]}>:</Label>
+    </Container>
+  )
+
+  const renderTimer = () => {
+    /*
+    caculate minutes
+    */
+    const resultCaculatMinute = Math.floor(seconds / numberSecondPerMinute) % numberSecondPerMinute
+    const minute = resultCaculatMinute >= 0 ? resultCaculatMinute : 0
+    const textMinute = (
+      <Container style={elementStyle}>
+        <Label style={[CountDownTheme.textStyle, textStyle]}>{minute.toString().padStart(2, '0')}</Label>
+      </Container>
+    )
+    /*
+    caculate seconds
+    */
+    const resultCaculatSecond = Math.round(seconds % numberSecondPerMinute)
+    const second = resultCaculatSecond >= 0 ? resultCaculatSecond : 0
+    const textSecond = (
+      <Container style={elementStyle}>
+        <Label style={[CountDownTheme.textStyle, textStyle]}>{second.toString().padStart(2, '0')}</Label>
+      </Container>
+    )
+
+    let textDay: React.ReactNode | null = null
+    let textHour: React.ReactNode | null = null
+    /*
+    caculate day
+    */
+    if (FormatTime.ddhhmmss === FormatTime[format]) {
+      const resultCaculatDay = Math.floor(seconds / secondPerDay)
+      const day = resultCaculatDay >= 0 ? resultCaculatDay : 0
+      textDay = (
+        <Container style={elementStyle}>
+          <Label style={[CountDownTheme.textStyle, textStyle]}>{day.toString().padStart(2, '0')}</Label>
+        </Container>
+      )
+    }
+    /*
+    caculate hours
+    */
+    if (FormatTime.hhmmss === FormatTime[format] || FormatTime.ddhhmmss === FormatTime[format]) {
+      const resultCaculatHour = Math.floor(seconds / secondPerHour)
+      const hour = resultCaculatHour >= 0 ? resultCaculatHour % hourPerDay : 0
+      textHour = (
+        <Container style={elementStyle}>
+          <Label style={[CountDownTheme.textStyle, textStyle]}>{hour.toString().padStart(2, '0')}</Label>
+        </Container>
+      )
+    }
+
+    return (
+      <Container style={[CountDownTheme.containerStyle, containerStyle]}>
+        {textDay && textDay}
```

Comment:
should it convert to Boolean, because textDay has case = null, so if it return null it will return error string outside Text component